src/components/booksForm.js

```diff
@@ -1,52 +1,84 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import { CreateBook } from '../actions';
 import Categories from '../helper/constants';
 
-const BooksForm = ({ createBook }) => {
-  let title;
-  let category;
-
-  return (
-    <form
-      onSubmit={e => {
-        e.preventDefault();
-        createBook(
-          {
-            id: parseInt(100000 * Math.random(), 10),
-            title: title.value,
-            category: category.value,
-          },
-        );
-      }}
-    >
-      <div>
-        <label htmlFor="txtTitle">
-          Title
-          <input type="text" id="txtTitle" name="txtTitle" ref={input => { (title = input); }} />
-        </label>
-      </div>
-      <div>
-        <label htmlFor="selCategory">
-          Category
-          <select name="selCategory" id="selCategory" ref={select => { (category = select); }}>
-            {
-              Categories.map(item => (
-                <option key={item} value={item}>{item}</option>
-              ))
-            }
-          </select>
-        </label>
-      </div>
-      <div>
-        <button type="submit">
-          Add Book
-        </button>
-      </div>
-    </form>
-  );
-};
+class BooksForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: '',
+    };
+    this.title = '';
+    this.category = '';
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange() {
+    this.setState({
+      title: this.title.value,
+      category: this.category.value,
+    });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+
+    const { title, category } = this.state;
+    const { createBook } = this.props;
+
+    if (title.trim().length === 0) {
+      return;
+    }
+    if (category.includes('Select')) {
+      return;
+    }
+
+    createBook(
+      {
+        id: parseInt(100000 * Math.random(), 10),
```

Comment:
* [ ] Let us use a more efficient way to create our Id generator, we can use packages such as `shortid` or `uuid`.